src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java

```diff
@@ -227,28 +215,38 @@ public static void assignRequest(final IStandardRequestManager manager, final IR
             currentResolver = ResolverHandler.getResolverForRequest(manager, request);
         }
 
+        IToken<?> parent = null;
+        if (request.hasParent())
+        {
+            parent = request.getParent();
+        }
+
         //Cancel the request to restart the search
         processInternalCancellation(manager, request.getToken());
 
         if (currentResolver != null)
         {
-            if (manager.getResolverRequestMap().containsKey(currentResolver.getRequesterId()))
+            if (manager.getRequestResolverRequestAssignmentDataStore().getAssignments().containsKey(currentResolver.getRequesterId()))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)